packages/jsx/src/utils.ts

```diff
@@ -0,0 +1,43 @@
+import { atom, Atom, AtomMaybe, CtxSpy, isAtom } from '@reatom/core'
+import { isObject} from '@reatom/utils'
+
+type ClassValue = AtomMaybe<
+  | string
+  | number
+  | boolean
+  | null
+  | undefined
+  | Array<ClassValue>
+  | Atom<ClassValue>
+  | Record<string, AtomMaybe<string | number | boolean | null | undefined | object>>
+  | ((ctx: CtxSpy) => ClassValue)
+>
+
+export const cn = (value: ClassValue): Atom<string> => atom((ctx) => reatomClassName(ctx, value))
+
+const reatomClassName = (ctx: CtxSpy, value: ClassValue): string => {
+  let className = ''
+  while (isAtom(value)) value = ctx.spy(value)
+  if (typeof value === 'string') {
+    className = value
+  } else if (typeof value === 'function') {
+    className = reatomClassName(ctx, value(ctx))
+  } else if (Array.isArray(value)) {
+    const length = value.length
+    for (let i = 0; i < length; i++) {
+      const normalized = reatomClassName(ctx, value[i])
+      if (normalized) {
```

Comment:
Always use strict comparison in this kind utilities to improve performance
```suggestion
      if (normalized !== '') {
```